pkg/dns/azure/client/auth.go

```diff
@@ -132,3 +102,88 @@ func endpointToScope(endpoint string) string {
 	}
 	return scope
 }
+
+// azureCreds uses the Azure SDK for Go's standard default credential chain.It attempts to authenticate with each of
+// these credential types, in the following order, stopping when one provides a token:
+//   - [EnvironmentCredential]
+//   - [WorkloadIdentityCredential], if environment variable configuration is set by the Azure workload
+//     identity webhook. Use [WorkloadIdentityCredential] directly when not using the webhook or needing
+//     more control over its configuration.
+//   - [ManagedIdentityCredential]
+//   - [AzureCLICredential]
+func azureCreds(options *azidentity.DefaultAzureCredentialOptions) (*azidentity.DefaultAzureCredential, error) {
+	err := os.Setenv("AZURE_CLIENT_SEND_CERTIFICATE_CHAIN", "true")
+	if err != nil {
+		return nil, fmt.Errorf("could not set AZURE_CLIENT_SEND_CERTIFICATE_CHAIN")
+	}
+
+	if certPath := os.Getenv("AZURE_CLIENT_CERTIFICATE_PATH"); certPath != "" {
+		// Set up a watch on our config file; if it changes, we should exit -
+		// (we don't have the ability to dynamically reload config changes).
+		stopCh := make(chan struct{})
```

Comment:
Wiring up `stopCh` is the missing piece, and it's is the reason why this is still a draft PR—right?

The whole "watch for a change and terminate the process if the change is observed" pattern makes this feel like something that might fit into library-go, similar to the featuregate logic: https://github.com/openshift/cluster-ingress-operator/blob/4975cde4f58e993bf8a1a4f72951036c2715b6b4/pkg/operator/operator.go#L110-L111

Have you considered putting some of this logic in library-go, or is that not practical given time constraints or other constraints?

Reply:
I believe I wired up the `stopCh` appropriately based on another OpenShift repo this code is from.

Yeah Alberto mentioned it would be nice to have this logic in library-go somewhere but given time constraints to do that, it's not practical at the moment.